### PR TITLE
JIT: Fix double handling of call operands when importing intrinsic

### LINKED ISSUE #96461: JIT assert on Unix in `ReturnTypeDesc::InitializeStructReturnType`
The new logic in https://github.com/dotnet/runtime/pull/96455#issuecomment-1875921135 is triggering an assert on OSX/Linux:
```
Assert failure(PID 10027 [0x0000272b], Thread: 85988 [0x14fe4]): Assertion failed '!m_inited' in 'System.Runtime.Intrinsics.VectorMath:Log2Single[System.Runtime.Intrinsics.Vector128`1[float],System.Runtime.Intrinsics.Vector128`1[int],System.Runtime.Intrinsics.Vector128`1[uint]](System.Runtime.Intrinsics.Vector128`1[float]):System.Runtime.Intrinsics.Vector128`1[float]' during 'Importation' (IL size 976; hash 0xefb066b7; Tier0)
```

The failure is possibly related to the use of generics + static virtuals in interfaces + the vector types being HVAs on these platforms.

### PR DESCRIPTION
**Pull request by jakobbotsch**
Some intrinsics can be a noop when they are imported. For example,
`Unsafe.BitCast<T, T>(X)` just returns its operand when importing. If
that operand is a call then we will run the tail logic of call
importation for that call node twice, since that logic tries to filter
out intrinsic importation with an `IsCall()` check. Switch the check to
be based on `!bIntrinsicImported` instead.

Fix #96461

### CODE REVIEW

**Comment by jakobbotsch** [comment]
cc @dotnet/jit-contrib PTAL @AndyAyersMS 

[No diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=516288&view=ms.vss-build-web.run-extensions-tab)

File: src/coreclr/jit/importercalls.cpp [hunk 1]
```diff
@@ -1303,11 +1303,10 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
             eeGetCallSiteSig(pResolvedToken->token, pResolvedToken->tokenScope, pResolvedToken->tokenContext, sig);
         }
 
-        typeInfo tiRetVal = verMakeTypeInfo(sig->retType, sig->retTypeClass);
-
-        if (call->IsCall())
+        // Sometimes "call" is not a GT_CALL (if we imported an intrinsic that didn't turn into a call)
+        if (!bIntrinsicImported)
```
Comment (tannergooding) [inline src/coreclr/jit/importercalls.cpp:1307]: Is this something that needs backported?

Looks like it's been a potential issue for a long time, although it's unclear to me why we're only just seeing this "now" given how widely used intrinsics have been, including in potentially tail recursive positions, and we've many that can be no-ops or just return one of the input operands (such as `op_UnaryPlus`, `AsInt32`, etc).

-- Notably looks like there are some asserts in this block checking that `!bIntrinsicImported` that can potentially be removed.

Comment (AndyAyersMS) [reply src/coreclr/jit/importercalls.cpp:1307]: Seems likely this double-processing ended up with the same result, so didn't actually cause a bug.

**Review by AndyAyersMS — APPROVED** [review]

File: src/coreclr/jit/importercalls.cpp (hunk 1 above)
Comment (jakobbotsch) [reply src/coreclr/jit/importercalls.cpp:1307]: I don't think it needs to be backported, I think it's very hard for the code that runs twice here to cause problems in release builds. Also we need a customer report for backporting.

> Looks like it's been a potential issue for a long time, although it's unclear to me why we're only just seeing this "now" given how widely used intrinsics have been, including in potentially tail recursive positions, and we've many that can be no-ops or just return one of the input operands (such as op_UnaryPlus, AsInt32, etc).

Presumably people do not write code that involves noop intrinsics like `op_UnaryPlus` very often, and even more rarely that also satisfies the other conditions to hit the bug.

> -- Notably looks like there are some asserts in this block checking that !bIntrinsicImported that can potentially be removed.

Let me address that as part of a future PR to rerunning CI.

**Pull request merged**